tests/validators/string.test.ts

```diff
@@ -96,4 +96,208 @@ describe('StringValidator', () => {
+				});
+
+				test.each(['google.com', 'foo.bar'])('GIVEN %s THEN throws a ConstraintError', (input) => {
+					expect(() => urlPredicate.parse(input)).toThrow(new ConstraintError('s.string.url', 'Invalid URL', input, 'expected.url'));
+				});
+			});
+
+			describe('With protocol', () => {
+				const urlPredicateWithProtocol = s.string.url({ allowedProtocols: ['git:'] });
+
+				test.each(['git://foo.bar'])('GIVEN %s THEN returns given value', (input) => {
+					expect(urlPredicateWithProtocol.parse(input)).toBe(input);
+				});
+
+				test.each(['https://google.com', 'http://foo.bar'])('GIVEN %s THEN throws a ConstraintError', (input) => {
+					expect(() => urlPredicateWithProtocol.parse(input)).toThrow(
+						new ConstraintError('s.string.url', 'Invalid URL protocol', input, 'expected to be one of: git:')
+					);
+				});
+			});
+
+			describe('With domain', () => {
+				const urlPredicateWithDomain = s.string.url({ allowedDomains: ['google.com'] });
+
+				test.each(['https://google.com', 'http://google.com'])('GIVEN %s THEN returns given value', (input) => {
+					expect(urlPredicateWithDomain.parse(input)).toBe(input);
+				});
+
+				test.each(['https://foo.bar', 'http://foo.bar'])('GIVEN %s THEN throws a ConstraintError', (input) => {
+					expect(() => urlPredicateWithDomain.parse(input)).toThrow(
+						new ConstraintError('s.string.url', 'Invalid URL domain', input, 'expected to be one of: google.com')
+					);
+				});
+			});
+		});
+
+		describe('uuid', () => {
+			const uuid5s = ['2a7ff881-2944-55ae-94b0-b2ed34432297'];
+			const uuid4s = ['450d6a23-9e6f-45d9-9d5a-fd4f6e014f16'];
+			const uuid3s = ['2962d7f2-92f2-3105-8606-2234808bdfc8'];
+			const uuid1s = ['c310deb0-9785-11ec-8e65-592cb74aa664'];
+			const nullUuid = '00000000-0000-0000-0000-000000000000';
+
+			describe('uuid5', () => {
+				const uuid5Predicate = s.string.uuid({ version: 5 });
+
+				test.each(uuid5s)('GIVEN %s THEN returns given value', (input) => {
+					expect(uuid5Predicate.parse(input)).toBe(input);
+				});
+
+				test.each(['1b671a64-40d5-491e-99b0-da01ff1f3342'])('GIVEN %s THEN throws a ConstraintError', (input) => {
+					expect(() => uuid5Predicate.parse(input)).toThrow(
+						new ConstraintError('s.string.uuid', 'Invalid string format', input, 'expected UUID v5')
+					);
+				});
+			});
+
+			describe('uuid4', () => {
+				const uuid4Predicate = s.string.uuid({ version: 4 });
+
+				test.each(uuid4s)('GIVEN %s THEN returns given value', (input) => {
+					expect(uuid4Predicate.parse(input)).toBe(input);
+				});
+
+				test.each(['6e8bc430-9a1b-4f7f-b7a5', '6e8bc430-9a1b-4f7f-b7a5', '450d6a23-9e6f-35d9-9d5a-fd4f6e014f16'])(
+					'GIVEN %s THEN throws a ConstraintError',
+					(input) => {
+						expect(() => uuid4Predicate.parse(input)).toThrow(
+							new ConstraintError('s.string.uuid', 'Invalid string format', input, 'expected UUID v4')
+						);
+					}
+				);
+
+				test('Given no version then match v4', () => {
```

Comment:
What's the meaning of this test?

Reply:
If we don't specify a version then it'll check for uuid4.